### PR TITLE
Fix CVE-2023-44483 Vulnerability in Ballerina SOAP Module

### PR DESCRIPTION
**Pull request by Nuvindu**
## Purpose

This aims at resolving [CVE-2023-44483](https://santuario.apache.org/secadv.data/CVE-2023-44483.txt.asc?version=1&modificationDate=1697782758000&api=v2), a security vulnerability that affects the Apache Santuario - XML Security for Java library, which is utilized by the Ballerina SOAP module as an external dependency. The vulnerability could potentially lead to private key disclosure when generating an XML Signature and enabling debug-level logging.

To fix this vulnerability, the Apache Santuario - XML Security for Java library is updated to version 3.0.3. This update will ensure the the security of the Ballerina SOAP module.

### CODE REVIEW

File: ballerina/Ballerina.toml [hunk 1]
```diff
@@ -37,5 +37,5 @@ path = "./lib/wss4j-ws-security-common-3.0.1.jar"
 [[platform.java17.dependency]]
 groupId = "org.apache.santuario"
```
Comment (dilanSachi) [inline ballerina/Ballerina.toml:38]: Dont we have a Ballerina.toml in build-config folder?

File: native/build.gradle [hunk 2]
```diff
@@ -30,7 +30,7 @@ dependencies {
     implementation group: 'org.ballerinalang', name: 'ballerina-runtime', version: "${ballerinaLangVersion}"
     implementation 'com.google.guava:guava:32.1.1-jre'
     implementation 'org.apache.wss4j:wss4j-ws-security-dom:3.0.1'
-    implementation 'org.apache.santuario:xmlsec:3.0.2'
+    implementation 'org.apache.santuario:xmlsec:3.0.3'
```
Comment (dilanSachi) [inline native/build.gradle:33]: better to not hardcode the versions. check how this is done in other modules - https://github.com/ballerina-platform/module-ballerina-grpc/blob/master/native/build.gradle#L39

File: ballerina/Ballerina.toml (hunk 1 above)
Comment (Nuvindu) [reply ballerina/Ballerina.toml:38]: There, it takes the dependency version from the gradle.properties file

File: native/build.gradle (hunk 2 above)
Comment (Nuvindu) [reply native/build.gradle]: Fixed in [8b1d56d](https://github.com/ballerina-platform/module-ballerina-soap/pull/113/commits/8b1d56d868b6eddcae4db67524ead4f4ca321689)

File: ballerina/Ballerina.toml (hunk 1 above)
Comment (dilanSachi) [reply ballerina/Ballerina.toml:38]: Ack. Forgot that it has only placeholders

**Review by dilanSachi — APPROVED** [review]

**Pull request merged**